### DISCUSSION TITLE
Section 1 NFT

### DISCUSSION THREAD

**Discussion by allwin199** [Cyfrin/security-and-auditing-full-course-s23#117] [Q&A] [answered]
```solidity
@param the function selector of the first one you need to call
@param the abi encoded data... hint! Use chisel to figure out what to use here...
@param yourTwitterHandle - Your twitter handle. Can be a blank string.

function solveChallenge(bytes4 selectorOne, bytes memory inputData, string memory yourTwitterHandle) external {
        (bool successOne, bytes memory responseDataOne) = i_helperContract.call(abi.encodeWithSelector(selectorOne));
        if (!successOne || uint256(bytes32((responseDataOne))) != 1) {
            revert S1__WrongSelector();
        }

        (bool successTwo, bytes memory responseDataTwo) = i_helperContract.call(inputData);
        if (!successTwo || uint256(bytes32((responseDataTwo))) != 1) {
            revert S1__WrongData();
        }
        _updateAndRewardSolver(yourTwitterHandle);
    }

```

My doubt is for which function I should get the function selector.
Also, what should the input data that should be turned into bytes?

Thanks in Advance.

ps: To understand how to get the solution is my motivation not just to get the NFT.

**Comment by allwin199** [comment, marked as answer]
Answered by "@equious.eth" in discord

**Reply by PatrickAlphaC** [reply]
What was the answer? Could you put it here?

**Reply by allwin199** [reply]
I completely missed the part of `i_helperContract`

Whenever `solveChallenge` is called. It makes a call to the `i_helperContract `

Inside the `i_helperContract` there is a function called `returnTrue`

we need to get the function selector for `returnTrue`.

While calling `solveChallenge` give the function selector of `returnTrue` as input args to `selectorOne ` and `inputData `

**Reply by PatrickAlphaC** [reply]
nice work!
